Library/Homebrew/test/cleanup_spec.rb

```diff
@@ -141,5 +166,106 @@
 
       expect(npm_cache).not_to exist
     end
+
+    it "cleans up all files and directories" do
+      git = (HOMEBREW_CACHE/"gist--git")
+      gist = (HOMEBREW_CACHE/"gist")
+      svn = (HOMEBREW_CACHE/"gist--svn")
+
+      git.mkpath
+      gist.mkpath
+      FileUtils.touch svn
+
+      allow(ARGV).to receive(:value).with("prune").and_return("all")
+
+      described_class.cleanup_cache
+
+      expect(git).not_to exist
+      expect(gist).to exist
+      expect(svn).not_to exist
+    end
+
+    it "does not clean up directories that are not VCS checkouts" do
+      git = (HOMEBREW_CACHE/"git")
+      git.mkpath
+      allow(ARGV).to receive(:value).with("prune").and_return("all")
+
+      described_class.cleanup_cache
+
+      expect(git).to exist
+    end
+
+    it "cleans up VCS checkout directories with modified time < prune time" do
+      foo = (HOMEBREW_CACHE/"--foo")
+      foo.mkpath
+      allow(ARGV).to receive(:value).with("prune").and_return("1")
+      allow_any_instance_of(Pathname).to receive(:mtime).and_return(Time.now - 60 * 60 * 24 * 2)
+      described_class.cleanup_cache
+      expect(foo).not_to exist
+    end
+
+    it "does not clean up VCS checkout directories with modified time >= prune time" do
+      foo = (HOMEBREW_CACHE/"--foo")
+      foo.mkpath
+      allow(ARGV).to receive(:value).with("prune").and_return("1")
+      described_class.cleanup_cache
+      expect(foo).to exist
+    end
+
+    context "cleans old files in HOMEBREW_CACHE" do
+      let(:bottle) { (HOMEBREW_CACHE/"testball-0.0.1.bottle.tar.gz") }
+      let(:testball) { (HOMEBREW_CACHE/"testball-0.0.1") }
+
+      before(:each) do
+        FileUtils.touch(bottle)
+        FileUtils.touch(testball)
+        (HOMEBREW_CELLAR/"testball"/"0.0.1").mkpath
+        FileUtils.touch(CoreTap.instance.formula_dir/"testball.rb")
+      end
+
+      after(:each) do
+        FileUtils.rm_rf(bottle)
+        FileUtils.rm_rf(testball)
+        FileUtils.rm_rf(HOMEBREW_CELLAR/"testball")
+        FileUtils.rm_rf(CoreTap.instance.formula_dir/"testball.rb")
```

Comment:
For example: https://github.com/Homebrew/brew/blob/2c43e069c25e3c7cc255f5cd105df3597246b952/Library/Homebrew/test/spec_helper.rb#L110-L129